contracts/oracle/GOhmEthOracle.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";
+import {Decimal, IOracle} from "./IOracle.sol";
+import {CoreRef} from "../refs/CoreRef.sol";
+
+interface IgOHM {
+    function mint(address _to, uint256 _amount) external;
+
+    function burn(address _from, uint256 _amount) external;
+
+    function index() external view returns (uint256);
+
+    function balanceFrom(uint256 _amount) external view returns (uint256);
+
+    function balanceTo(uint256 _amount) external view returns (uint256);
+
+    function migrate(address _staking, address _sOHM) external;
+}
+
+/// @title GOHM Eth Oracle
+/// @notice Report the gOHM price in units of ETH. Calculated by reading a Chainlink OHM V2
+///         oracle price and multiplying that by the queried OHM index
+contract GOhmEthOracle is IOracle, CoreRef {
+    using Decimal for Decimal.D256;
+
+    /// @notice the referenced chainlink oracle
+    AggregatorV3Interface public immutable chainlinkOHMETHOracle;
+
+    /// @notice Oracle decimals normalizer
+    uint256 public oracleDecimalsNormalizer;
+
+    /// @notice gOHM token address.
+    IgOHM public constant GOHM = IgOHM(0x0ab87046fBb341D058F17CBC4c1133F25a20a52f);
+
+    /// @param _core Fei Core for reference
+    /// @param _chainlinkOHMETHOracle Chainlink OHM V2 oracle reporting in terms of ETH
+    constructor(address _core, address _chainlinkOHMETHOracle) CoreRef(_core) {
+        chainlinkOHMETHOracle = AggregatorV3Interface(_chainlinkOHMETHOracle);
+        _init();
+    }
+
+    // @dev: decimals of the oracle are expected to never change, if Chainlink
+    // updates that behavior in the future, we might consider reading the
+    // oracle decimals() on every read() call.
+    function _init() internal {
+        uint8 oracleDecimals = chainlinkOHMETHOracle.decimals();
+        oracleDecimalsNormalizer = 10**uint256(oracleDecimals);
+    }
+
+    /// @notice Update the oracle price
+    /// No-op, Chainlink is updated automatically. Price is multiplied by OHM index at runtime
+    function update() external override whenNotPaused {}
+
+    /// @notice determine if read value is stale
+    /// @return true if read value is stale
+    function isOutdated() external view override returns (bool) {
+        (uint80 roundId, , , , uint80 answeredInRound) = chainlinkOHMETHOracle.latestRoundData();
+        return answeredInRound != roundId;
+    }
+
+    /// @notice Determine the gOHM price. Report in units of ETH.
+    /// @return oracle gOHM price in units of ETH
+    /// @return true if price is valid
+    function read() external view override returns (Decimal.D256 memory, bool) {
+        (uint80 roundId, int256 ohmEthPrice, , , uint80 answeredInRound) = chainlinkOHMETHOracle.latestRoundData();
+        bool valid = !paused() && ohmEthPrice > 0 && answeredInRound == roundId;
```

Comment:
Should add in a check for how recent the latest round was.

Reply:
I'm not sure this is necessary unless there's a strong reason. The trigger conditions for this data feed on Chainlink are a price deviation of 2% or 86400 seconds have passed since the last update. 

This oracle is only used to value the gOHM in the PCV - it's not a significant chunk, 0.8% - so don't think we need to be more defensive

Reply:
The trigger conditions specify when an update *should* happen, but don't guarantee that it will. We should be checking how old the latest update is, and mark the update as invalid if it's too old.